### DISCUSSION TITLE
how to find keyring in docker-compose wallet?

### DISCUSSION THREAD

**Discussion by dckc** [Agoric/agoric-sdk#5789] [Q&A] [answered]
In discord, Zerek asked how to find it; they gave some motivation:
> for custody companies who have an offline signature process this is very useful to get easy access these information to match if everything is correct

for reference: https://github.com/Agoric/agoric-sdk/wiki/Setting-up-an-Agoric-Dapp-Client-with-docker-compose

**Comment by dckc** [comment]
I think I found the answer; details are below. @tgrecojs maybe you can explain?

```
$ AG_SOLO_BASEDIR=/tmp/solo123 ~/projects/agoric-sdk/packages/solo/bin/ag-solo setup --netconfig=${NETCONFIG_URL:-https://devnet.agoric.net/network-config}
...
ctrl-c
...

$ agd --home=/tmp/solo123/ag-cosmos-helper-statedir --keyring-backend=test keys list
- name: ag-solo
  type: local
  address: agoric1xm0yqt0ylvppsutz8ezggnhyvd2rufkwp5dkj8
  pubkey: '{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"A4o8sfsY2DdX7L5/XkWpKvU7ccwydX4zAgYaz6c/1Eaf"}'
  mnemonic: ""
```

also note in `docker-compose.yml`

```
    environment:
      - "AG_SOLO_BASEDIR=/state/${SOLO_HOME:-202205280523443a4bb7}"
```

**Reply by Ekzer** [reply]
That's interesting, thank you for this first answer

**Reply by Ekzer** [reply]
So I guess we could put somewhere in doc that --home option is $AG_SOLO_BASEDIR/ag-cosmos-helper-statedir and needs --keyring-backed=test option as well

**Comment by Ekzer** [comment, marked as answer]
I finally found how to retrieve private key.
To precise something first : 
`--keyring-backend=test` means that the privateKey is password less. According to cosmos documentation https://docs.cosmos.network/master/run-node/keyring.html#the-test-backend

When you want to get private key, you have 2 ways : 

`agd keys export ag-solo --home=/state/202205280523443a4bb7/ag-cosmos-helper-statedir --keyring-backend=test --unarmored-hex --unsafe` will export to HEX format 

OR

`agd keys export ag-solo --home=/state/202205280523443a4bb7/ag-cosmos-helper-statedir --keyring-backend=test` will ask for passphrase and then will generate an encrypted version
